src/vs/workbench/parts/git/browser/gitWidgets.ts

```diff
@@ -100,14 +100,17 @@ export class GitStatusbarItem implements IStatusbarItem {
 
 	private onGitServiceChange(): void {
 		const model = this.gitService.getModel();
+		const ps1 = model.getPS1();
+		const tags = model.getRefs().filter(iref => iref.commit.substr(0,8) === ps1);
+		const name = tags.length > 0 ? tags[0].name : model.getPS1();
```

Comment:
This breaks the `+*!` symbols in front of branches, since whenever a branch is checked out, `model.getPS1` is never called.